test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -109,310 +113,386 @@ func (s *snapshottableTestSuite) DefineTests(driver TestDriver, pattern testpatt
+			var (
+				vs        *unstructured.Unstructured
+				vscontent *unstructured.Unstructured
+				vsc       *unstructured.Unstructured
+			)
+
+			ginkgo.BeforeEach(func() {
+				sr := CreateSnapshotResource(sDriver, config, pattern, pvc.GetName(), pvc.GetNamespace())
+				vs = sr.Vs
+				vscontent = sr.Vscontent
+				vsc = sr.Vsclass
+				cleanupSteps = append(cleanupSteps, func() {
+					framework.ExpectNoError(sr.CleanupResource())
+				})
+			})
+
+			ginkgo.It("should delete the VolumeSnapshotContent according to its deletion policy", func() {
+				err = DeleteAndWaitSnapshot(dc, vs.GetNamespace(), vs.GetName(), framework.Poll, framework.SnapshotDeleteTimeout)
+				framework.ExpectNoError(err)
+
+				switch pattern.SnapshotDeletionPolicy {
+				case testpatterns.DeleteSnapshot:
+					ginkgo.By("checking the SnapshotContent has been deleted")
+					err = utils.WaitForGVRDeletion(dc, SnapshotContentGVR, vscontent.GetName(), framework.Poll, framework.SnapshotDeleteTimeout)
+					framework.ExpectNoError(err)
+				case testpatterns.RetainSnapshot:
+					ginkgo.By("checking the SnapshotContent has not been deleted")
+					err = utils.WaitForGVRDeletion(dc, SnapshotContentGVR, vscontent.GetName(), 1*time.Second /* poll */, 30*time.Second /* timeout */)
+					framework.ExpectError(err)
+				}
+			})
+			ginkgo.It("should create snapshot objects correctly", func() {
+				ginkgo.By("checking the snapshot")
+				// Get new copy of the snapshot
+				vs, err = dc.Resource(SnapshotGVR).Namespace(vs.GetNamespace()).Get(context.TODO(), vs.GetName(), metav1.GetOptions{})
+				framework.ExpectNoError(err)
+
+				// Get the bound snapshotContent
+				snapshotStatus := vs.Object["status"].(map[string]interface{})
+				snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+				vscontent, err = dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+
+				snapshotContentSpec := vscontent.Object["spec"].(map[string]interface{})
+				volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+
+				// Check SnapshotContent properties
+				ginkgo.By("checking the SnapshotContent")
+				framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], vsc.GetName())
+				framework.ExpectEqual(volumeSnapshotRef["name"], vs.GetName())
+				framework.ExpectEqual(volumeSnapshotRef["namespace"], vs.GetNamespace())
+			})
+			ginkgo.It("should restore from snapshot with saved data after modifying source data", func() {
+				var restoredPVC *v1.PersistentVolumeClaim
+				var restoredPod *v1.Pod
+				modifiedMntTestData := fmt.Sprintf("modified data from %s namespace", pvc.GetNamespace())
+
+				ginkgo.By("modifying the data in the source PVC")
+
+				command := fmt.Sprintf("echo '%s' > /mnt/test/data", modifiedMntTestData)
+				RunInPodWithVolume(cs, pvc.Namespace, pvc.Name, "pvc-snapshottable-data-tester", command, config.ClientNodeSelection)
```

Comment:
Cant we use the pod "pvc-snapshottable-tester" created in init() to modify the contents of the PV?
Here we are starting yet another pod "pvc-snapshottable-data-tester" to modify the pv contents.
If we re-use the "pvc-snapshottable-tester" the test timings would be less. Thoughts?

Reply:
Is there an easy way to talk to the process in that init pod? IIUC it needs to wait until after the snapshot is made before modifying the volume. 

Your point about reducing the timing is good, though, considering all the timeouts we've been having recently. Maybe it would be worth it to create some utility pod infrastructure that could easily communicate back and forth with the test to avoid starting up extra pods.

(all of that out of scope for this PR of course :-)

Reply:
We have some tests that do a Pod exec in order to modify data. That may be a method to consider if we don't want to launch a new pod.

Reply:
The pod "pvc-snapshottable-tester" created in init() is destroyed in init(), although there's no reason we can't keep it around so that we can reuse it. It looks like it currently takes ten seconds for a pod to become ready, and we would save ten seconds.

I was having trouble finding a way to execute commands in an existing pod, perhaps I can use this https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L500 and just ignore the error because I'm not looking for a string in output.